### DISCUSSION TITLE
Please kindly guide me how to add custom component into this react-markdown

### DISCUSSION THREAD

**Discussion by codeleeks** [remarkjs/remark#1328] [Q&A]
i'm new on `react-markdown`

**i want to transform jsx like strings to react component without mdx.
i just wanted to use this `react-markdown` library and related plugins.**

here's a example

This is my markdown source.(.md file)
![image](https://github.com/remarkjs/remark/assets/166087781/30da6c25-5a27-4c4c-9bfb-3095af6fd9d2)

i want to transform it into react component like below
![image](https://github.com/remarkjs/remark/assets/166087781/9b87b92f-7f24-4dd1-b3c3-17e877482f6e)

my code is
```javascript
      <Markdown
        rehypePlugins={[rehypeHighlight, rehypeRaw, rehypeSlug, remarkGfm]}
        className='contents'
        components={{
          code(props) {
            const { children, className, node, ...rest } = props
            const match = /^hljs\s/.test(className)
            return (
              <code {...rest} className={match ? className : 'hljs'}>
                {children}
              </code>
            )
          },
         // this is what i've been trying to add.
          messagebox(props) {
            const { children, node, ...rest } = props
            return <MessageBox>{children}</MessageBox>
          },
        }}
      >
        {skipped}
      </Markdown>
```

but this code doesn't work very well. 
please ignore korean. 

### Expect: 
![image](https://github.com/remarkjs/remark/assets/166087781/eb90b514-398e-4512-aae1-27c5502effe6)

### Actual:
![image](https://github.com/remarkjs/remark/assets/166087781/2d611eab-9c16-4f15-b16a-0546425d13d3)

i guess that's because following contents became children of my custom component.

how should i fix this?

**Comment by wooorm** [comment]
You have MDX. Use MDX.

**Comment by codeleeks** [comment]
> You have MDX. Use MDX.

is there any workaround to fix this issue using `react-markdown`, not MDX

**Reply by wooorm** [reply]
Anything is possible but you are asking the wrong questions.
Why do you not want to use `@mdx-js/mdx` when you have MDX?
